.github/workflows/gradle-check.yml

```diff
@@ -27,8 +27,27 @@ jobs:
         with:
           ref: ${{ github.event.pull_request.head.sha }}
 
+      - uses: tj-actions/changed-files@v44
+        id: changed-files
+
+      - id: run-gradle-check
+        env:
+          ALL_CHANGED_FILES: ${{ steps.changed-files.outputs.all_changed_files }}
+        run: |
+          run_gradle_check=false
+
+          for filename in ${ALL_CHANGED_FILES}; do
+            # Run gradle check if file ends in [java/gradle/json]
```

Comment:
I like the spirit here but there are many files that can fail the build that don't meet these extensions:

- no extension, e.g. https://github.com/opensearch-project/OpenSearch/blob/main/libs/core/src/main/resources/META-INF/services/org.opensearch.core.compress.spi.CompressorProvider
- License files (.txt I think?) are checked as a part of the build
- gradle.properties
- probably many many others?

Reply:
We can do an inverse set where we skip gradle check if the changes are limited to certain file types

Reply:
If going this route, I'd rather do a strictly positive list. If the commits only affects: [CHANGELOG*.md, README.md, ...] then skip

Reply:
👍 to @peternied 's suggestion, other examples are: version.properties, bwcVersions, ....

Reply:
That sounds good. I was just bouncing the idea across.
I'll work on refining and testing this out. 
Thanks folks.

Reply:
I find [paths-ignore](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-excluding-paths) as an easier option. We can keep on adding to the list and also manage file-types, directories, etc. This won't even initiate the workflow run for ignored paths. Example: https://github.com/opensearch-project/OpenSearch/pull/13477

Reply:
That sounds good. I'll close out this PR and we can keep adding to `paths-ignore`